### PR TITLE
Improving AVR memory map and giving support for more CPU parameters.

### PR DESCRIPTION
**Pull request by killabytenow**
This is an experimental branch for improving support for several CPU's.
Also I am improving the AVR memory map, for emulating better the real arch.

The main reason for this PR is to pass the Travolis checks. If everything runs well, maybe I will ask for a merge some commits later.

pupu

### CODE REVIEW

**Comment by radare** [comment]
```
aac
axt 0x4ec

[35mDiff: [0m--- /tmp/r2-regressions//search-ViUzzH/exp	2016-10-27 18:44:30.925405804 +0000
+++ /tmp/r2-regressions//search-ViUzzH/out	2016-10-27 18:44:30.989408108 +0000
@@ -1 +1 @@
-call 0x3e2 call fcn.000004ec in fcn.00000394
+call 0x3e2 call fcn.000004ec in fcn.00000360

```

**Comment by radare** [comment]
ping

File: libr/anal/p/anal_avr.c [hunk 1]
```diff
@@ -345,13 +410,13 @@ INST_HANDLER (cbi) {	// CBI A, b
...
 
 	// read port a and clear bit b
-	io_port = __generic_io_dest (a, 0);
+	io_port = __generic_io_dest (a, 0, cpu);
 	ESIL_A ("0xff,%d,1,<<,^,%s,&,", b, io_port);
 	r_strbuf_free (io_port);
 
 	// write result to port a
-	io_port = __generic_io_dest (a, 1);
-	ESIL_A ("%s,=,", r_strbuf_get (io_port));
```
Comment (j123123) [inline libr/anal/p/anal_avr.c]: what meaning have this ``=,`` in esil and why remove it?

**Comment by alvarofe** [comment]
rebase cause the test that @radare pasted i think it was problem in master due to tinyrange change

**Comment by radare** [comment]
i guess that we can merge this

**Pull request merged**